app/workers/daily_grouping_updater.rb

```diff
@@ -0,0 +1,27 @@
+class DailyGroupingUpdater
+  include Sidekiq::Worker
+
+  def initialize
+    @system_account = Watcher.find_by_email("kairos-robot@omadahealth.com")
+    PaperTrail.whodunnit = @system_account
+  end
+
+  def perform(app_name: nil)
+    raise "An app_name must be specified." unless app_name.present?
```

Comment:
Instead of defaulting the app_name parameter to nil and raising an error if app_name is nil you could could make app_name a non-default/required parameter ;)